go/vt/vtgate/evalengine/func.go

```diff
@@ -256,18 +325,29 @@ func builtinFuncCollation(env *ExpressionEnv, args []EvalResult, result *EvalRes
 	})
 }
 
-func builtinFuncIsNull(_ *ExpressionEnv, args []EvalResult, result *EvalResult) {
+func (builtinCollation) typeof(_ *ExpressionEnv, args []Expr) (sqltypes.Type, flag) {
 	if len(args) != 1 {
-		throwArgError("ISNULL")
+		throwArgError("COLLATION")
 	}
+	return sqltypes.VarChar, 0
+}
+
+type builtinIsNull struct{}
```

Comment:
Nice. I'll do that at the `sqlparser` level :+1:

Reply:
Done, I did the rewrite in the evalengine because the `isnull` is not a keyword nor reserved word in MySQL, which means that they're also doing the rewrite in their eval engine. :smile: